### ISSUE TITLE
how to convert eva model

### ISSUE THREAD

**Issue by rico227** [w1oves/Rein#5]
There is a script to convert dino backbone. How do I have to adapt the following function to use EVA02-B?

def load_backbone(path: str):
    if not osp.isfile(path):
        raise FileNotFoundError(
            f"{path} dont exist(absolute path: {osp.abspath(path)})"
        )
    weight = torch.load(path, map_location="cpu")
    weight["pos_embed"] = torch.cat(
        (
            weight["pos_embed"][:, :1, :],
            F.interpolate(
                weight["pos_embed"][:, 1:, :]
                .reshape(1, 37, 37, 1024)
                .permute(0, 3, 1, 2),
                size=(32, 32),
                mode="bicubic",
                align_corners=False,
            )
            .permute(0, 2, 3, 1)
            .reshape(1, 1024, 1024),
        ),
        dim=1,
    )
    weight["patch_embed.proj.weight"] = F.interpolate(
        weight["patch_embed.proj.weight"].float(),
        size=(16, 16),
        mode="bicubic",
        align_corners=False,
    )
    return weight

**Comment by w1oves** [comment]
You can convert weights in a similar manner. I'm currently occupied, and I'll provide the relevant code next week.

**Comment by w1oves** [comment]
Thank you for your attention! I have uploaded the code for converting EVA02 weights in tools/convert_models. Please download pre-trained weights from [EVA02-L](https://huggingface.co/Yuxin-CV/EVA-02/blob/main/eva02/pt/eva02_L_pt_m38m_p14to16.pt) and [EVA02-B](https://huggingface.co/Yuxin-CV/EVA-02/blob/main/eva02/pt/eva02_B_pt_in21k_p14to16.pt).

**Comment by yasserben** [comment]
Hi ! Thank you for sharing the script with us. However, I have 2 questions :

1- The second link you provided for EVA02-B points to eva02_L_pt_in21k_p14to16.pt which if I'm not mistaken is still an EVA02-L but pre-trained on ImageNet-21K
2- Do we really need the interpolation, since in both the link you provide, the patch embeddings are already of size 16x16 hence the "p14tpp16.pt" in the name of the checkpoints ?

Thank you !!

**Comment by w1oves** [comment]
> Hi ! Thank you for sharing the script with us. However, I have 2 questions :
> 
> 1- The second link you provided for EVA02-B points to eva02_L_pt_in21k_p14to16.pt which if I'm not mistaken is still an EVA02-L but pre-trained on ImageNet-21K
> 2- Do we really need the interpolation, since in both the link you provide, the patch embeddings are already of size 16x16 hence the "p14tpp16.pt" in the name of the checkpoints ?
> 
> Thank you !! 

Sorry for providing the wrong link earlier, I have already updated the correct link in the corresponding issue. Regarding interpolation, it is indeed unnecessary; however, it does not cause any errors, so the script remains correct.
